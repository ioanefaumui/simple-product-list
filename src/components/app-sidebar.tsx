import {
  Sidebar,
  SidebarContent,
  SidebarGroup,
  SidebarGroupContent,
  SidebarGroupLabel,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
} from "./ui/sidebar";

import { Boxes, Calendar, Home, Inbox, Search, Settings } from "lucide-react";

const items = [
  {
    title: "Produtos",
    url: "#",
    icon: Boxes,
  },
];

export function AppSidebar() {
  return (
    <Sidebar>
      <SidebarContent>
        <SidebarGroup>
          <SidebarGroupLabel>Application</SidebarGroupLabel>
          <SidebarGroupContent>
            <SidebarMenu>
              {items.map((item) => (
                <SidebarMenuItem key={item.title}>
                  <SidebarMenuButton asChild>
                    <a href={item.url}>
                      <item.icon />
                      <span>{item.title}</span>
                    </a>
                  </SidebarMenuButton>
                </SidebarMenuItem>
              ))}
            </SidebarMenu>
          </SidebarGroupContent>
        </SidebarGroup>
      </SidebarContent>
    </Sidebar>
  );
}
