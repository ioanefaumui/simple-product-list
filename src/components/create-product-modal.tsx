"use client";

import { useEffect, useState } from "react";
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogHeader,
  DialogTitle,
} from "./ui/dialog";
import { Button } from "./ui/button";
import { Label } from "@radix-ui/react-dropdown-menu";
import { Input } from "./ui/input";
import { ProductService } from "@/services";
import { Dices } from "lucide-react";
import { toast } from "sonner";

export function CreateProductModal() {
  const [open, setOpen] = useState(false);
  const [errors, setErrors] = useState<Record<string, string>>({});
  const [form, setForm] = useState({
    name: "",
    category: "",
    amount: "",
    description: "",
    image: "",
  });

  useEffect(() => {
    if (!open) {
      setForm({
        name: "",
        category: "",
        amount: "",
        description: "",
        image: "",
      });
      setErrors({});
    }
  }, [open]);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;

    if (value) {
      setErrors({ ...errors, [name]: "" });
    }

    if (name === "amount") {
      const raw = value.replace(/\D/g, "");
      const numeric = Number(raw) / 100;

      const formatted = new Intl.NumberFormat("pt-BR", {
        style: "currency",
        currency: "BRL",
      }).format(numeric);

      setForm((prev) => ({ ...prev, amount: formatted }));
    } else {
      setForm((prev) => ({ ...prev, [name]: value }));
    }
  };

  function parseCurrencyBR(value: string): number {
    return Number(value.replace(/[^\d,.-]/g, "").replace(",", "."));
  }

  const handleSubmit = async () => {
    const productService = new ProductService();
    const newErrors: Record<string, string> = {};

    if (!form.name) newErrors.name = "Nome é obrigatório";
    if (!form.category) newErrors.category = "Categoria é obrigatória";
    if (!form.amount) newErrors.amount = "Preço é obrigatório";
    if (!form.description) newErrors.description = "Descrição é obrigatória";
    if (!form.image) newErrors.image = "Imagem é obrigatória";

    if (Object.keys(newErrors).length > 0) {
      setErrors(newErrors);
      return;
    }

    setErrors({});

    try {
      await productService.create({
        name: form.name,
        category: form.category,
        amount: parseCurrencyBR(form.amount),
        description: form.description,
        image: form.image,
      });
      setOpen(false);
      toast(`Produto criado com sucesso!`);
      setForm({
        name: "",
        category: "",
        amount: "",
        description: "",
        image: "",
      });
    } catch (err) {
      console.error("Erro ao criar produto:", err);
      toast(`Houve um erro ao criar o produto.`);
    }
  };

  const gerarImagemAleatoria = () => {
    const id = Math.floor(Math.random() * 200) + 1;
    const url = `https://picsum.photos/id/${id}/300/200`;

    setForm((prev) => ({ ...prev, image: url }));
  };

  return (
    <Dialog open={open} onOpenChange={setOpen}>
      <Button className="ml-auto" onClick={() => setOpen(true)}>
        Adicionar produto
      </Button>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Novo Produto</DialogTitle>
          <DialogDescription>
            Preencha os dados para cadastrar.
          </DialogDescription>
        </DialogHeader>

        <div className="space-y-3">
          <div>
            <Label>Nome</Label>
            <Input
              name="name"
              aria-invalid={!!errors.name}
              value={form.name}
              onChange={handleChange}
              autoComplete="off"
              className="aria-[invalid='true']:border-destructive"
            />
            {errors.name && (
              <p className="text-sm text-destructive mt-1">{errors.name}</p>
            )}
          </div>
          <div>
            <Label>Categoria</Label>
            <Input
              name="category"
              aria-invalid={!!errors.category}
              value={form.category}
              autoComplete="off"
              onChange={handleChange}
            />
            {errors.category && (
              <p className="text-sm text-destructive mt-1">{errors.category}</p>
            )}
          </div>
          <div>
            <Label>Preço</Label>
            <Input
              name="amount"
              aria-invalid={!!errors.amount}
              value={form.amount || ""}
              onChange={handleChange}
              autoComplete="off"
              type="text"
              inputMode="numeric"
            />
            {errors.amount && (
              <p className="text-sm text-destructive mt-1">{errors.amount}</p>
            )}
          </div>
          <div>
            <Label>Descrição</Label>
            <Input
              name="description"
              aria-invalid={!!errors.description}
              value={form.description}
              autoComplete="off"
              onChange={handleChange}
            />
            {errors.description && (
              <p className="text-sm text-destructive mt-1">
                {errors.description}
              </p>
            )}
          </div>
          <div>
            <Label>Imagem (URL)</Label>
            <div className="relative">
              <Input
                name="image"
                aria-invalid={!!errors.image}
                value={form.image}
                onChange={handleChange}
                autoComplete="off"
                type="text"
                className="pr-10"
              />
              <button
                type="button"
                onClick={gerarImagemAleatoria}
                className="absolute right-2 top-1/2 -translate-y-1/2 text-gray-500 hover:text-black"
              >
                <Dices size={16} />
              </button>
            </div>
            {errors.image && (
              <p className="text-sm text-destructive mt-1">{errors.image}</p>
            )}
          </div>
          <Button className="w-full mt-4" onClick={handleSubmit}>
            Cadastrar
          </Button>
        </div>
      </DialogContent>
    </Dialog>
  );
}
