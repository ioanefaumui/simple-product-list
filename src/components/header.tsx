import { useEffect, useState } from "react";
import { SidebarTrigger } from "./ui/sidebar";

export function Header() {
  return (
    <header className="flex items-center p-2 w-full sticky top-0 border-b">
      <SidebarTrigger />
    </header>
  );
}
