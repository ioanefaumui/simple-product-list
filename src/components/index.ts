export * from "./header";
export * from "./app-sidebar";
export * from "./product-list";
export * from "./product-pagination";
