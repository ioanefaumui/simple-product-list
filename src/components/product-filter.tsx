"use client";

import {
  Popover,
  PopoverContent,
  PopoverTrigger,
} from "@/components/ui/popover";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { useRouter, useSearchParams } from "next/navigation";
import { useState, useEffect } from "react";
import { Filter } from "lucide-react";
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from "@/components/ui/select";
import { Label } from "@radix-ui/react-dropdown-menu";

export function ProductFilter() {
  const searchParams = useSearchParams();
  const router = useRouter();

  const [open, setOpen] = useState(false);
  const [min, setMin] = useState("");
  const [max, setMax] = useState("");
  const [orderPrice, setOrderPrice] = useState<"asc" | "desc" | undefined>(
    undefined,
  );

  useEffect(() => {
    setMin(searchParams.get("preco_min") || "");
    setMax(searchParams.get("preco_max") || "");
    setOrderPrice(
      searchParams.get("ordenar_preco") === "desc"
        ? "desc"
        : searchParams.get("ordenar_preco") === "asc"
          ? "asc"
          : undefined,
    );
  }, [searchParams]);

  const hasFilters =
    !!searchParams.get("preco_min") ||
    !!searchParams.get("preco_max") ||
    !!searchParams.get("ordenar_preco");

  const handleApply = () => {
    const params = new URLSearchParams(searchParams);

    if (min) params.set("preco_min", min);
    else params.delete("preco_min");

    if (max) params.set("preco_max", max);
    else params.delete("preco_max");

    if (orderPrice) {
      params.set("ordenar_preco", orderPrice);
    } else {
      params.delete("ordenar_preco");
    }

    params.set("pagina", "1");

    router.push(`?${params.toString()}`);
    setOpen(false);
  };

  const handleClear = () => {
    const params = new URLSearchParams(searchParams);
    params.delete("preco_min");
    params.delete("preco_max");
    params.delete("ordenar_preco");
    params.delete("pagina");

    setMin("");
    setMax("");
    setOrderPrice(undefined);
    router.push(`?${params.toString()}`);
    setOpen(false);
  };

  return (
    <Popover open={open} onOpenChange={setOpen}>
      <PopoverTrigger asChild>
        <Button
          variant="outline"
          className="relative flex items-center gap-2 w-fit"
        >
          <Filter />
          <span className="hidden sm:inline-block">Filtros</span>
          {hasFilters && (
            <span className="absolute -top-1 -right-1 size-2 rounded-full bg-red-600" />
          )}
        </Button>
      </PopoverTrigger>
      <PopoverContent className="w-64 space-y-4 mr-4 ">
        <div>
          <Label className="text-sm mb-1">Preço mínimo</Label>
          <Input
            value={min}
            onChange={(e) => setMin(e.target.value)}
            placeholder="Ex: 100"
            type="number"
          />
        </div>
        <div>
          <Label className="text-sm mb-1">Preço máximo</Label>
          <Input
            value={max}
            onChange={(e) => setMax(e.target.value)}
            placeholder="Ex: 500"
            type="number"
          />
        </div>
        <div>
          <Label className="text-sm mb-1">Ordenar por preço</Label>
          <Select
            value={orderPrice}
            onValueChange={(val) => setOrderPrice(val as "asc" | "desc")}
          >
            <SelectTrigger className="w-full">
              <SelectValue placeholder="Ordenar por" />
            </SelectTrigger>
            <SelectContent>
              <SelectItem value="asc">Mais baratos</SelectItem>
              <SelectItem value="desc">Mais caros</SelectItem>
            </SelectContent>
          </Select>
        </div>
        <div className="flex flex-col gap-2">
          <Button className="w-full" onClick={handleApply}>
            Aplicar filtros
          </Button>
          <Button variant="outline" className="w-full" onClick={handleClear}>
            Limpar
          </Button>
        </div>
      </PopoverContent>
    </Popover>
  );
}
