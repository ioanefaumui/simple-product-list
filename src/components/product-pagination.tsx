"use client";

import { useSearchParams } from "next/navigation";
import { Pagination, PaginationContent, PaginationItem } from "./ui/pagination";
import { PaginationMeta } from "@/services";
import Link from "next/link";
import { Button } from "./ui/button";

interface Props {
  meta: Omit<PaginationMeta, "data"> | null;
}

export function ProductPagination({ meta }: Props) {
  if (!meta) return null;

  const searchParams = useSearchParams();
  const currentPage = Number(searchParams.get("pagina") || "1");

  const buildHref = (newPage: number) => {
    const params = new URLSearchParams(searchParams);
    params.set("pagina", String(newPage));
    return `?${params.toString()}`;
  };

  const pagesToShow = 3;
  const pages: (number | "...")[] = [];

  const start = Math.max(1, currentPage - 1);
  const end = Math.min(meta.pages, start + pagesToShow - 1);

  if (start > 1) {
    pages.push(1);
    if (start > 2) pages.push("...");
  }

  for (let i = start; i <= end; i++) {
    pages.push(i);
  }

  if (end < meta.pages) {
    if (end < meta.pages - 1) pages.push("...");
    pages.push(meta.pages);
  }

  return (
    <Pagination className="mt-4">
      <PaginationContent>
        <PaginationItem className="hidden lg:inline-block">
          <Button disabled={!meta.prev} variant="ghost" className="relative">
            <Link
              scroll={false}
              href={buildHref(meta.prev ?? 0)}
              className="absolute inset-0"
            />
            Anterior
          </Button>
        </PaginationItem>

        {pages.map((page, index) => (
          <PaginationItem key={index}>
            {page === "..." ? (
              <Button variant="ghost" disabled>
                ...
              </Button>
            ) : (
              <Link href={buildHref(page)} scroll={false}>
                <Button variant={page === currentPage ? "outline" : "ghost"}>
                  {page}
                </Button>
              </Link>
            )}
          </PaginationItem>
        ))}

        <PaginationItem className="hidden lg:inline-block">
          <Button disabled={!meta.next} variant="ghost" className="relative">
            <Link
              scroll={false}
              href={buildHref(meta.next ?? 0)}
              className="absolute inset-0"
            />
            Pr√≥xima
          </Button>
        </PaginationItem>
      </PaginationContent>
    </Pagination>
  );
}
