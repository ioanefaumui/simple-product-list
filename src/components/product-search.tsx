"use client";

import { useState, useEffect } from "react";
import { Input } from "./ui/input";
import { Button } from "./ui/button";
import { Search, X } from "lucide-react";
import { useSearchParams, useRouter } from "next/navigation";

export function ProductSearch() {
  const [value, setValue] = useState("");
  const searchParams = useSearchParams();
  const router = useRouter();

  useEffect(() => {
    const initial = searchParams.get("nome") || "";
    setValue(initial);
  }, [searchParams]);

  const buildSearchURL = () => {
    const params = new URLSearchParams(searchParams);
    if (value.trim()) {
      params.set("nome", value.trim());
      params.set("pagina", "1");
    } else {
      params.delete("nome");
    }
    return `?${params.toString()}`;
  };

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    router.push(buildSearchURL());
  };

  const handleClear = () => {
    const params = new URLSearchParams(searchParams);
    params.delete("nome");
    params.set("pagina", "1");
    setValue("");
    router.push(`?${params.toString()}`);
  };

  return (
    <form
      onSubmit={handleSubmit}
      className="flex items-center gap-2 w-full lg:max-w-sm"
    >
      <div className="relative w-full">
        <Input
          placeholder="Buscar produto..."
          value={value}
          onChange={(e) => setValue(e.target.value)}
          className="pr-10"
        />
        {value && (
          <button
            type="button"
            onClick={handleClear}
            className="absolute right-2 top-1/2 -translate-y-1/2 text-gray-500 hover:text-black"
          >
            <X size={16} />
          </button>
        )}
      </div>

      <Button type="submit">
        <Search />
      </Button>
    </form>
  );
}
