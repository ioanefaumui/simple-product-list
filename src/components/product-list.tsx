"use client";

import * as React from "react";
import {
  ColumnDef,
  ColumnFiltersState,
  SortingState,
  VisibilityState,
  flexRender,
  getCoreRowModel,
  getFilteredRowModel,
  getPaginationRowModel,
  getSortedRowModel,
  useReactTable,
} from "@tanstack/react-table";
import { MoreHorizontal } from "lucide-react";

import { Button } from "@/components/ui/button";
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuLabel,
  DropdownMenuSeparator,
  DropdownMenuTrigger,
} from "@/components/ui/dropdown-menu";
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "@/components/ui/table";
import Image from "next/image";
import { ProductRead } from "@/types";
import { PaginationMeta, ProductService } from "@/services";
import { ProductPagination } from "./product-pagination";
import { ProductSearch } from "./product-search";
import { ProductFilter } from "./product-filter";
import { CreateProductModal } from "./create-product-modal";
import { toast } from "sonner";
import { useRouter } from "next/navigation";

export const columns: ColumnDef<ProductRead>[] = [
  {
    accessorKey: "image",
    header: "Foto",
    cell: ({ row }) => (
      <div className="relative w-10 h-8">
        <Image
          src={row.getValue("image")}
          alt={row.getValue("name")}
          fill
          sizes="(max-width: 640px) 100vw, 40px"
          className="object-cover"
        />
      </div>
    ),
  },
  {
    accessorKey: "name",
    header: "Nome",
    cell: ({ row }) => (
      <div className="capitalize min-w-[7.5rem]">{row.getValue("name")}</div>
    ),
  },
  {
    accessorKey: "description",
    header: "Descrição",
    cell: ({ row }) => (
      <div className="min-w-[12.5rem]">{row.getValue("description")}</div>
    ),
  },
  {
    accessorKey: "category",
    header: "Categoria",
    cell: ({ row }) => (
      <div className="capitalize">{row.getValue("category")}</div>
    ),
  },
  {
    accessorKey: "amount",
    header: () => <div className="text-right">Preço</div>,
    cell: ({ row }) => {
      const amount = parseFloat(row.getValue("amount"));
      const formatted = new Intl.NumberFormat("pt-BR", {
        style: "currency",
        currency: "BRL",
      }).format(amount);
      return <div className="text-right font-medium">{formatted}</div>;
    },
  },
  {
    id: "actions",
    enableHiding: false,
    cell: ({ row }) => {
      const product = row.original;
      const router = useRouter();
      const handleDelete = async () => {
        const confirm = window.confirm(`Deseja remover "${product.name}"?`);
        if (!confirm) return;

        const productService = new ProductService();
        try {
          await productService.delete(product.id);
          toast.success("Produto removido com sucesso");
          router.refresh();
        } catch (err) {
          console.error(err);
          toast.error("Erro ao remover produto");
        }
      };
      return (
        <DropdownMenu>
          <DropdownMenuTrigger asChild>
            <Button variant="ghost" className="h-8 w-8 p-0">
              <span className="sr-only">Abrir menu</span>
              <MoreHorizontal />
            </Button>
          </DropdownMenuTrigger>
          <DropdownMenuContent align="end">
            <DropdownMenuLabel>Ações</DropdownMenuLabel>
            <DropdownMenuItem onClick={() => alert(product.name)}>
              Editar
            </DropdownMenuItem>
            <DropdownMenuSeparator />
            <DropdownMenuItem onClick={handleDelete}>Remover</DropdownMenuItem>
          </DropdownMenuContent>
        </DropdownMenu>
      );
    },
  },
];

interface Props {
  products: ProductRead[];
  meta: Omit<PaginationMeta, "data"> | null;
}

export function ProductList({ products = [], meta }: Props) {
  const [sorting, setSorting] = React.useState<SortingState>([]);
  const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>(
    [],
  );
  const [columnVisibility, setColumnVisibility] =
    React.useState<VisibilityState>({});

  const table = useReactTable({
    data: products,
    columns,
    onSortingChange: setSorting,
    onColumnFiltersChange: setColumnFilters,
    getCoreRowModel: getCoreRowModel(),
    getPaginationRowModel: getPaginationRowModel(),
    getSortedRowModel: getSortedRowModel(),
    getFilteredRowModel: getFilteredRowModel(),
    onColumnVisibilityChange: setColumnVisibility,
    state: {
      sorting,
      columnFilters,
      columnVisibility,
    },
  });

  return (
    <div className="w-full">
      <div className="flex flex-col lg:flex-row gap-x-2 gap-y-4 mt-4">
        <div className="flex gap-2 w-full">
          <ProductSearch />
          <ProductFilter />
        </div>
        <div className="col-span-2 text-right">
          <CreateProductModal />
        </div>
      </div>

      <div className="rounded-md border mt-4">
        <Table>
          <TableHeader>
            {table.getHeaderGroups().map((headerGroup) => (
              <TableRow key={headerGroup.id}>
                {headerGroup.headers.map((header) => {
                  return (
                    <TableHead key={header.id}>
                      {header.isPlaceholder
                        ? null
                        : flexRender(
                            header.column.columnDef.header,
                            header.getContext(),
                          )}
                    </TableHead>
                  );
                })}
              </TableRow>
            ))}
          </TableHeader>
          <TableBody>
            {table.getRowModel().rows?.length ? (
              table.getRowModel().rows.map((row) => (
                <TableRow
                  key={row.id}
                  data-state={row.getIsSelected() && "selected"}
                >
                  {row.getVisibleCells().map((cell) => (
                    <TableCell key={cell.id} className="whitespace-normal">
                      {flexRender(
                        cell.column.columnDef.cell,
                        cell.getContext(),
                      )}
                    </TableCell>
                  ))}
                </TableRow>
              ))
            ) : (
              <TableRow>
                <TableCell
                  colSpan={columns.length}
                  className="h-24 text-center"
                >
                  Nenhum resultado.
                </TableCell>
              </TableRow>
            )}
          </TableBody>
        </Table>
      </div>
      <div className="flex items-center justify-end space-x-2 py-4">
        <ProductPagination meta={meta} />
      </div>
    </div>
  );
}
