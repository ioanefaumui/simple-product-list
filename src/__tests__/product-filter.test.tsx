import { expect, test, vi } from "vitest";
import { render } from "@testing-library/react";
import { ProductFilter } from "@/components/product-filter";

vi.mock("next/navigation", () => ({
  useRouter: () => ({
    push: vi.fn(),
    refresh: vi.fn(),
    replace: vi.fn(),
    back: vi.fn(),
  }),
  useSearchParams: () => new URLSearchParams(""),
}));

test("product-filter matches snapshot", () => {
  const { baseElement } = render(<ProductFilter />);
  expect(baseElement).toMatchSnapshot();
});
