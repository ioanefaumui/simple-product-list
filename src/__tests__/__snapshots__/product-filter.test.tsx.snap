// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`product-filter matches snapshot 1`] = `
<body>
  <div>
    <button
      aria-controls="radix-«r0»"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="justify-center whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 relative flex items-center gap-2 w-fit"
      data-slot="popover-trigger"
      data-state="closed"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-funnel"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 20a1 1 0 0 0 .553.895l2 1A1 1 0 0 0 14 21v-7a2 2 0 0 1 .517-1.341L21.74 4.67A1 1 0 0 0 21 3H3a1 1 0 0 0-.742 1.67l7.225 7.989A2 2 0 0 1 10 14z"
        />
      </svg>
      <span
        class="hidden sm:inline-block"
      >
        Filtros
      </span>
    </button>
  </div>
</body>
`;
